tests/src/availability/healthcheck_availability_during_deployment_test.go

```diff
@@ -0,0 +1,114 @@
+package availability_test
+
+import (
+	"fmt"
+	"time"
+
+	vegeta "github.com/tsenart/vegeta/lib"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	availabilityTestRate             = 10
+	availabilityTestMaxDuration      = 2 * time.Hour
+	availabilityTestMaxLatency       = 300 * time.Millisecond
+	availabilityTestMaxErrors        = 0
+	availabilitySuccessRateThreshold = 99.9
+
+	vegetaRunForever = 0 * time.Second
+	vegetaKeepAlive  = true
+)
+
+func getAppsDomainZoneName() string {
+	return getConfigFromEnvironment("APPS_DNS_ZONE_NAME")
+}
+
+func getResourceVersion() string {
+	return getConfigFromEnvironment("PIPELINE_TRIGGER_VERSION")
+}
+
+func loadTest(appUri string, endpoint string, rate uint64) (*vegeta.Attacker, <-chan *vegeta.Result) {
+	targeter := vegeta.NewStaticTargeter(vegeta.Target{
+		Method: "GET",
+		URL:    appUri,
+	})
+
+	attacker := vegeta.NewAttacker(vegeta.KeepAlive(vegetaKeepAlive))
+	res := attacker.Attack(targeter, rate, vegetaRunForever)
+
+	return attacker, res
+}
+
+func errorRateThreshold(metrics vegeta.Metrics, minimumDuration time.Duration, errorRate float64) bool {
+	// metrics.Close() does trigger the computation of metrics, but does not stop any process
+	metrics.Close()
+
+	fmt.Printf(
+		" - Duration: %s, Requests: %d, Non 200 Requests: %d, Success Rate: %.2f%%\n",
+		metrics.Duration,
+		metrics.Requests,
+		metrics.Requests-uint64(metrics.StatusCodes["200"]),
+		metrics.Success*100,
+	)
+
+	if metrics.Duration > minimumDuration {
+		if metrics.Success < (1 - errorRate) {
+			return true
+		}
+	}
+	return false
+}
+
+var _ = Describe("Availability test", func() {
+
+	var metrics vegeta.Metrics
+	var stopAttackCriteria func() bool
+
+	Context("when runs (until the deployment is finished or error rate > 50%)", func() {
+		var attacker *vegeta.Attacker
+		var resultChannel <-chan *vegeta.Result
+
+		BeforeEach(func() {
+			stopAttackCriteria = func() bool {
+				if deploymentHasFinishedInConcourse() {
+					return true
+				}
+				if errorRateThreshold(metrics, 15*time.Second, 0.5) {
+					return true
+				}
+				return false
+			}
+		})
+
+		It("does not get reuest success rate less than 99.9%", func() {
```

Comment:
Can the test description use `availabilitySuccessRateThreshold` in case we change one value and forget to the update the other?